### PR TITLE
Get log for each attempt.

### PR DESCRIPTION
**Pull request by Binyang2014**
Change log API from  `/jobs/{job-name}/pods/podUid/logs` to 
 `/jobs/{job-name}/taskRoles/{taskRoleName}/taskIndex/{taskIndex}/attempts/{taskAttemptId}/logs`
 
Previous, get log for each task attempt need to query DB: 
1. get job level info. 2. get job attempt level info 3. get task level info 4. get task attempt level info. 

After this change, only need to get task level info.

### CODE REVIEW

**Review by yqwang-ms — APPROVED** [review]

File: src/rest-server/src/models/v2/job/log.js [hunk 1]
```diff
@@ -36,38 +36,45 @@ const loginLogManager = async (nodeIp, username, password) => {
...
-    if (!status) {
-      logger.error(`Failed to find pod which has pod uid ${podUid}`);
-      throw noPodLogsErr;
-    }
-    nodeIp = status.containerIp;
-    taskRoleName = key;
+  const taskStatus = taskDetail.data.attempts[Number(taskAttemptId)];
+  if (!taskStatus) {
+    logger.error(`Failed to find task to retrive log`);
+    throw noPodLogsErr;
```
Comment (suiguoxin) [inline src/rest-server/src/models/v2/job/log.js:64]: NoTaskLogErr ?

File: src/rest-server/src/models/v2/task.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ const get = async (frameworkName, jobAttemptIndex, taskRoleName, taskIndex) => {
   );
   if (taskRoleStatus) {
     taskStatus = taskRoleStatus.taskStatuses.find(
-      (taskRoleStatus) => taskRoleStatus.index === taskIndex,
+      (taskAttemptStatus) => taskAttemptStatus.index === taskIndex,
```
Comment (suiguoxin) [inline src/rest-server/src/models/v2/task.js:82]: Thx for the correction. This should be named `taskStatus`

**Review by debuggy — APPROVED** [review]

File: src/rest-server/src/models/v2/job/log.js (hunk 1 above)
Comment (Binyang2014) [reply src/rest-server/src/models/v2/job/log.js]: Fxied

File: src/rest-server/src/models/v2/task.js (hunk 2 above)
Comment (Binyang2014) [reply src/rest-server/src/models/v2/task.js]: Fixed

**Review by suiguoxin — APPROVED** [review]

**Pull request merged**